cmd/performance-profile-creator/cmd/root.go

```diff
@@ -114,60 +118,100 @@ func getNodeSelectorFromLabelSelector(labelSelector *metav1.LabelSelector) *v1.N
-			return nil, fmt.Errorf("failed to list mustGatherPath directories: %v", err)
-		}
-		for _, node := range nodes {
-			nodeName := node.Name()
-			node, err := clientset.CoreV1Interface.Nodes().Get(context.TODO(), nodeName, metav1.GetOptions{})
-			if err != nil {
-				return nil, err
-			}
-			workerNodes = append(workerNodes, node)
-		}
+	defer src.Close()
+
+	dec := k8syaml.NewYAMLOrJSONDecoder(src, 1024)
+	if err := dec.Decode(&node); err != nil {
+		return nil, fmt.Errorf("Error opening %q: %v", path, err)
 	}
-	return workerNodes, nil
+
+	return &node, nil
 }
 
-func getMCPByName(name string, clientset *ClientSet) (*machineconfigv1.MachineConfigPool, error) {
+func getNodeByName(mustGatherDirPath, nodeName string) (*v1.Node, error) {
+	var node *v1.Node
+	nodePathSuffix := path.Join(ClusterScopedResources, CoreNodes, nodeName)
+	path, err := getMustGatherFullPaths(mustGatherDirPath, nodePathSuffix)
+	if err != nil {
+		return nil, fmt.Errorf("Error obtaining MachineConfigPool %s: %v", nodeName, err)
+	}
 
-	mcp, err := clientset.MachineConfigPools().Get(context.TODO(), name, metav1.GetOptions{})
-	return mcp, err
+	node, err = getNode(path)
+	if err != nil {
+		return nil, fmt.Errorf("Error obtaining MachineConfigPool %s: %v", node.GetName(), err)
+	}
+	return node, nil
 }
 
-// NewClientSet returns a *ClientBuilder with the given kubeconfig.
-func NewClientSet(kubeconfig string) *ClientSet {
-	var config *rest.Config
-	var err error
+func getNodeList(mustGatherDirPath string) ([]*v1.Node, error) {
+	machines := make([]*v1.Node, 0)
 
-	if kubeconfig == "" {
-		kubeconfig = os.Getenv("KUBECONFIG")
+	nodePathSuffix := path.Join(ClusterScopedResources, CoreNodes)
+	nodePath, err := getMustGatherFullPaths(mustGatherDirPath, nodePathSuffix)
+	if err != nil {
+		return nil, fmt.Errorf("Error obtaining Nodes: %v", err)
+	}
+	nodes, err := ioutil.ReadDir(nodePath)
+	if err != nil {
+		return nil, fmt.Errorf("failed to list mustGatherPath directories: %v", err)
+	}
+	for _, node := range nodes {
+		nodeName := node.Name()
+		node, err := getNodeByName(mustGatherDirPath, nodeName)
+		if err != nil {
+			return nil, fmt.Errorf("Error obtaining Nodes %s: %v", nodeName, err)
+		}
+		machines = append(machines, node)
 	}
+	return machines, nil
+}
+
+func getMCPByName(mustGatherDirPath, mcpName string) (*machineconfigv1.MachineConfigPool, error) {
+	var mcp *machineconfigv1.MachineConfigPool
 
-	if kubeconfig != "" {
-		glog.V(4).Infof("Loading kube client config from path %q", kubeconfig)
-		config, err = clientcmd.BuildConfigFromFlags("", kubeconfig)
-	} else {
-		glog.V(4).Infof("Using in-cluster kube client config")
-		config, err = rest.InClusterConfig()
+	mcpPathSuffix := path.Join(ClusterScopedResources, MCPools, mcpName+YAMLSuffix)
+	mcpPath, err := getMustGatherFullPaths(mustGatherDirPath, mcpPathSuffix)
+	if err != nil {
+		return nil, fmt.Errorf("Error obtaining MachineConfigPool %s: %v", mcpName, err)
 	}
+
+	mcp, err = getMCP(mcpPath)
```

Comment:
i'd just do `return getMCP(mcpPath)`. Which value we add in the return below?
Additionally: do we get MCPs by any other mean different than "by name"? If not we should squash together the two functions

Reply:
Good point, will squash together.